src/libs/location/src/character.cpp

```diff
@@ -1621,10 +1623,25 @@ void Character::Recoil()
         return;
     if (!GetLocation()->IsSwimming())
         return;
-    fgtSetType = fgt_recoil;
-    fgtSetIndex = 0;
-    recoilWait = 0.8f;
-    isTurnLock = false;
+    VDATA *res = nullptr;
+    res = core.Event("ChrCheckEnergy", "is", GetId(), "recoil");
+    int32_t isEnable = 1;
```

Comment:
`isEnabled` : )
also what is enabled? this is not obvious name

Reply:
Corsairs coding makes my English worse :)
`isEnabled` signals that current action is possible. Passed the energy test or is enabled by default. `isEnoughEnergy` isn't actual if event doesn't exist in scripts. Maybe `isActionPossible`?

Reply:
I like `isActionPossible`

Reply:
Done.